lib/factory_bot/decorator/invocation_tracker.rb

```diff
@@ -6,22 +6,11 @@ def initialize(component)
         @invoked_methods = []
       end
 
-      if ::Gem::Version.new(::RUBY_VERSION) >= ::Gem::Version.new("3.0")
-        def method_missing(name, *args, **kwargs, &block) # rubocop:disable Style/MissingRespondToMissing
-          @invoked_methods << name
-          super
-        end
-      elsif ::Gem::Version.new(::RUBY_VERSION) >= ::Gem::Version.new("2.7")
-        ruby2_keywords def method_missing(name, *args, &block) # rubocop:disable Style/MissingRespondToMissing
-          @invoked_methods << name
-          super
-        end
-      else
-        def method_missing(name, *args, &block) # rubocop:disable Style/MissingRespondToMissing
-          @invoked_methods << name
-          super
-        end
```

Comment:
The version above was correct, but it seemed a lot of duplication, so I used the simpler pattern here too.